api/v1/flow/flow.proto

```diff
@@ -258,6 +272,7 @@ message TCP {
 
 message IP {
     string source = 1;
+    string snatAs = 5;
```

Comment:
A few nits:
- Should this be snake case? Would be easier to read.
- The name is a bit unobvious to me. Maybe `source_nat` or `original_source` or something? (I'm not sure if this the IP pre or post translation, but if it's post translation it could be `new_source`).
- Should probably add a comment to describe the field's purpose.

Reply:
Hey @chancez thanks for chiming in, no worries this PR is in draft because it's on top of https://github.com/cilium/cilium/pull/31226 but I opened it anyway exactly to get that kind of feedback :pray:

* snake case: yes, actually that is what [the protobuf style guide recommends](https://protobuf.dev/programming-guides/style/#message-field-names) but for some reason I got mixed up and thought camel case was to be preferred, so I'll update.
* The name is not ideal I agree, I couldn't come up with something better so I'm open to suggestions. To clarify: in the SNAT case IP.source is the pre translation IP and `snatAs` the post-translation IP. With that in mind, `source_nat` is ambiguous and `original_source` confusing IMHO, but let's hear more opinions on this.
* comment: yes.

Reply:
Hmmm yeah. Is `post_translate_blahblah` too much? Naming things is hard.

Reply:
`source_xlated` or `source_xlated_as` ?

Reply:
`source_xlated` or `source_translated` seems decent